command/ssh.go

```diff
@@ -127,118 +147,309 @@ func (c *SSHCommand) Run(args []string) int {
 		// be used by the user (ACL enforcement), then user should see an
 		// error message accordingly.
 		c.Ui.Output(fmt.Sprintf("Vault SSH: Role: %q", role))
+		c.role = role
 	}
 
-	data := map[string]interface{}{
-		"username": username,
-		"ip":       ip.String(),
+	// If no mode was given, perform the old-school lookup. Keep this now for
+	// backwards-compatability, but print a warning.
+	//
+	// TODO: remove in 0.9.0, convert to validation error
+	if c.mode == "" {
+		c.Ui.Warn("" +
+			"WARNING: No -mode specified. Use -mode to tell Vault which ssh\n" +
+			"authentication mode to use. In the future, you will need to tell\n" +
+			"Vault which mode to use. For now, Vault will attempt to guess based\n" +
+			"on the API response.")
+		_, cred, err := c.generateCredential()
```

Comment:
Great point - I had totally glossed over that. What I've done in the interim:

1. Updated the message to make that clearer
1. Revoke the generated credential after we get the data we need from it

While these aren't perfect, I think it's the best API/behavior I can come up with before 0.9 when we make -mode and -role required flags. Thoughts?